### ISSUE TITLE
How to programmatically get a list of all dependency parse tags of a loaded model?

### ISSUE THREAD

**Issue by fhamborg** [explosion/spaCy#5135]
The info is statically described in the documentation, i.e., https://spacy.io/api/annotation#dependency-parsing, but what if one needs to get the list of all dependency parse tags programmatically, e.g., in case different models are employed in a system? Is there an internal structure (that probably comes with / is set when loading a model) that can be used?

**Comment by adrianeboyd** [comment]
For the core components, you can access `.labels`:

```
import spacy
nlp = spacy.load("en_core_web_sm")
assert nlp.parser.labels == ('ROOT', 'acl', 'acomp', 'advcl', 'advmod', 'agent', 'amod', 'appos', 'attr', 'aux', 'auxpass', 'case', 'cc', 'ccomp', 'compound', 'conj', 'csubj', 'csubjpass', 'dative', 'dep', 'det', 'dobj', 'expl', 'intj', 'mark', 'meta', 'neg', 'nmod', 'npadvmod', 'nsubj', 'nsubjpass', 'nummod', 'oprd', 'parataxis', 'pcomp', 'pobj', 'poss', 'preconj', 'predet', 'prep', 'prt', 'punct', 'quantmod', 'relcl', 'xcomp')
assert nlp.get_pipe("parser").labels == ('ROOT', 'acl', 'acomp', 'advcl', 'advmod', 'agent', 'amod', 'appos', 'attr', 'aux', 'auxpass', 'case', 'cc', 'ccomp', 'compound', 'conj', 'csubj', 'csubjpass', 'dative', 'dep', 'det', 'dobj', 'expl', 'intj', 'mark', 'meta', 'neg', 'nmod', 'npadvmod', 'nsubj', 'nsubjpass', 'nummod', 'oprd', 'parataxis', 'pcomp', 'pobj', 'poss', 'preconj', 'predet', 'prep', 'prt', 'punct', 'quantmod', 'relcl', 'xcomp')
```

Likewise `nlp.tagger` or `nlp.get_pipe("tagger")` and `nlp.entity` or `nlp.get_pipe("ner")`.

**Comment by fhamborg** [comment]
Awesome! Thanks a lot!

**Comment by fhamborg** [comment]
Strange, when I run this code I get an error:

```
import spacy

nlp = spacy.load("en_core_web_lg")
print(nlp.parser.labels)
```

will result in
```
/Users/felix/anaconda3/envs/ctsacuda/bin/python /Users/felix/IdeaProjects/newstsc/dependencyparser.py
Traceback (most recent call last):
  File "/Users/felix/IdeaProjects/newstsc/dependencyparser.py", line 16, in <module>
    print(nlp.parser.labels)
  File "pipes.pyx", line 1022, in genexpr
  File "pipes.pyx", line 1022, in genexpr
IndexError: list index out of range

Process finished with exit code 1
```

**Comment by adrianeboyd** [comment]
Hmm, I can't reproduce this. Could you provide more information about your environment (`spacy info --markdown`) and spacy model versions (`spacy validate`)?

**Comment by fhamborg** [comment]
Sure, please see below:

## Info about spaCy

* **spaCy version:** 2.1.0
* **Platform:** Darwin-19.3.0-x86_64-i386-64bit
* **Python version:** 3.7.6

```
(ctsacuda) ➜  ~ spacy validate
✔ Loaded compatibility table

====================== Installed models (spaCy v2.1.0) ======================
ℹ spaCy installation:
/Users/felix/anaconda3/envs/ctsacuda/lib/python3.7/site-packages/spacy

TYPE      NAME             MODEL            VERSION
package   en-core-web-lg   en_core_web_lg   2.1.0   ✔
```

**Comment by adrianeboyd** [comment]
That looks like a bug (#4275) that didn't get fixed until v2.2.0. Unless you have a particular reason for using v2.1, I'd recommending upgrading to a more recent version (v2.2+).

If you specifically need v2.1 (and even then I'd recommend upgrading to at least v2.1.8), you have to work around that bug, but the information is still available:

```python
labels = set()
for move in nlp.parser.move_names:
    if "-" in move:
        move = move.split("-")[1]
        if "||" in move:
            move = move.split("||")[1]
        labels.add(move)

assert labels == {'agent', 'acl', 'intj', 'predet', 'expl', 'prt', 'meta', 'attr', 'pcomp', 'parataxis', 'compound', 'auxpass', 'mark', 'prep', 'dative', 'xcomp', 'npadvmod', 'advcl', 'preconj', 'pobj', 'csubj', 'dep', 'punct', 'case', 'ROOT', 'quantmod', 'cc', 'det', 'subtok', 'neg', 'nmod', 'amod', 'advmod', 'nsubjpass', 'appos', 'conj', 'dobj', 'acomp', 'nsubj', 'csubjpass', 'nummod', 'ccomp', 'oprd', 'poss', 'relcl', 'aux'}
```
